designer/src/components/field-editor.tsx

```diff
@@ -331,6 +399,37 @@ export const FieldEditor = ({
             </Stack>
           </Grid>
         </Grid>
+        <Dialog
+          open={deleteWarningOpen}
+          onClose={() => setDeleteWarningOpen(false)}
+        >
+          <DialogTitle>Field Deletion Warning</DialogTitle>
+          <DialogContent>
+            <p>
+              This field is referenced in the following conditions. Deleting it
+              will invalidate them:
+            </p>
+            <ul>
+              {conditionsAffected.map((condition, index) => (
+                <li key={index}>{condition}</li>
+              ))}
+            </ul>
+            <p>Are you sure you want to proceed?</p>
+          </DialogContent>
+          <DialogActions>
+            <Button
+              onClick={e => {
+                e.stopPropagation();
+                setDeleteWarningOpen(false);
+              }}
+            >
+              Cancel
+            </Button>
+            <Button onClick={confirmDelete} color="error">
+              Delete Anyway
```

Comment:
I think I would prefer to forbid deleting the field if it is used in conditions - ask them to go delete the conditions first. Then we don't end up with invalid notebooks.